.github/workflows/pr-check.yaml

```diff
@@ -295,6 +295,9 @@ jobs:
     if: ${{ contains(github.event.pull_request.labels.*.name, 'area/ci') || !github.event.pull_request.draft }}
     name: smoke e2e tests
     runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      checks: write
```

Comment:
question: I don't understand how you can ask permissions to write in the PR check job where users have no rights ?

Reply:
Let me check if it is needed.

Reply:
Seems like with `annotate_only` it still works fine. removing `permissions` from all changes.